src/window.rs

```diff
@@ -493,24 +493,29 @@ impl Window {
         unsafe { ffi::gdk_window_get_root_coords(self.to_glib_none().0, x, y, root_x, root_y) }
     }
 
-    pub fn get_device_position(&self, device: &Device, x: &mut i32, y: &mut i32,
-        mask: &mut ::ModifierType) -> Option<Window> {
+    pub fn get_device_position(&self, device: &Device) -> (Option<Window>, i32, i32, ::ModifierType) {
         unsafe {
-            from_glib_none(ffi::gdk_window_get_device_position(
-                self.to_glib_none().0,
-                device.to_glib_none().0,
-                x, y, mask))
+            let mut x = mem::uninitialized();
+            let mut y = mem::uninitialized();
+            let mut mask = mem::uninitialized();
+            let ret = from_glib_none(ffi::gdk_window_get_device_position(self.to_glib_none().0,
+                                                                         device.to_glib_none().0,
+                                                                         &mut x, &mut y, &mut mask));
+            (ret, x, y, from_glib(mask))
         }
     }
 
     #[cfg(feature = "v3_10")]
-    pub fn get_device_position_double(&self, device: &Device, x: &mut f64, y: &mut f64,
-        mask: &mut ::ModifierType) -> Option<Window> {
+    pub fn get_device_position_double(&self, device: &Device) -> (Option<Window>, f64, f64, ::ModifierType) {
         unsafe {
-            from_glib_none(ffi::gdk_window_get_device_position_double(
-                self.to_glib_none().0,
-                device.to_glib_none().0,
-                x, y, mask))
+            let mut x = mem::uninitialized();
+            let mut y = mem::uninitialized();
+            let mut mask = mem::uninitialized();
+            let ret = from_glib_none(
+                ffi::gdk_window_get_device_position_double(self.to_glib_none().0,
+                                                           device.to_glib_none().0,
+                                                           &mut x, &mut y, &mut mask));
+            (ret, x, y, from_glib(mask))
```

Comment:
Yep, this seems to be more correct than before. I wonder if the generator would do the same...